packages/frontend/src/features/sqlRunner/hooks/useTableFields.tsx

```diff
@@ -0,0 +1,35 @@
+import { type ApiError, type ApiWarehouseTableFields } from '@lightdash/common';
+import { useQuery } from '@tanstack/react-query';
+import { lightdashApi } from '../../../api';
+
+export type GetTableFieldsParams = {
+    projectUuid: string;
+    tableName: string | undefined;
+};
+
+const fetchTableFields = async ({
+    projectUuid,
+    tableName,
+}: GetTableFieldsParams) =>
+    lightdashApi<ApiWarehouseTableFields>({
+        url: `/projects/${projectUuid}/sqlRunner/tables/${tableName}`,
+        method: 'GET',
+        body: undefined,
+    });
+
+export const useTableFields = ({
+    projectUuid,
+    tableName,
+}: GetTableFieldsParams) => {
+    return useQuery<ApiWarehouseTableFields, ApiError>({
+        queryKey: ['sqlRunner', 'tables', tableName, projectUuid],
+        queryFn: () =>
+            fetchTableFields({
+                projectUuid,
+                tableName,
+            }),
+        retry: false,
+        enabled: !!tableName,
+        // TODO: handle error on no schema found
```

Comment:
if the handling of `undefined` schema is OK, I think we should handle it at some point here...?

Reply:
afaik the schema is needed to get the tables first, in the backend, so you might not reach this state if that happens...

Reply:
removed comment